### PR TITLE
feat: let the editor handle any debug errors

### PR DESCRIPTION
**Pull request by kittaakos**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -108,34 +108,52 @@ export function activate(context: ExtensionContext) {
...
     try {
-        info = JSON.parse(rawStdout);
+        info = JSON.parse(stdout);
     } catch (err) {
-        console.error(`Could not parse JSON: <${rawStdout}>`);
-        showError(err);
+        throw err;
     }
     if (!info) {
         return false;
```
Comment (AlbyIanna) [inline src/extension.ts:159]: This code looks fine to me, but I would just like to suggest a little rearranging here to make it a bit more readable:
```ts
let info: DebugInfo | undefined = undefined;
    try {
        const args = ['debug', '-I', '-b', config.board.fqbn, config.sketchPath, '--format', 'json'];
        const { stdout, stderr } = await exec(config?.cliPath || '.', args);
        if (!stdout && stderr) {
            throw new Error(stderr);
        }
        info = JSON.parse(stdout);
        if (!info) {
            return false;
        }
    } catch (err) {
        throw err;
    }
```

I moved `stdout` and `stderr` inside the first `try{} catch{}`. That way we can avoid having another `try{} catch{}` block.

@kittaakos what do you think?

Comment (kittaakos) [reply src/extension.ts]: Good idea. Thank you!

Comment (kittaakos) [reply src/extension.ts]: 7ef1b58

**Review by AlbyIanna — APPROVED** [review]
Thank you for applying the change! LGTM 🚀

**Comment by kittaakos** [comment]
Great, thanks for your time! I need to modify this PR once more and drop a240d93, remove VSIX from version control, and do a release after the merge.

**Pull request merged**